ardupilot_cartographer/README.md

```diff
@@ -0,0 +1,147 @@
+# ardupilot_ros: ROS 2 use cases with Ardupilot
+
+[![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit)](https://github.com/pre-commit/pre-commit)
+
+## Requirements
+
+### System Requirements
+
+* [ROS Humble](https://docs.ros.org/en/humble/Installation.html)
+
+* [Gazebo Garden](https://gazebosim.org/docs/garden/install)
+
+* [Cartographer ROS](https://google-cartographer-ros.readthedocs.io/en/latest/)
+   * Recommended: Install Google Cartographer with rosdep
+
+### Workspace Requirements
+
+* [ardupilot_gz](https://github.com/ArduPilot/ardupilot_gz)
+
+* [cartographer_ros]()
+
+## Installation
+
+Clone this repository into your ros2 workspace alongside ardupilot_gz:
+```bash
+cd ~/ros2_ws/src
+git clone git@github.com:ardupilot/ardupilot_ros.git
+```
+
+Install dependencies using rosdep:
+```bash
+cd ~/ros2_ws
+rosdep install --from-paths src --ignore-src -r --skip-keys gazebo-ros-pkgs
+```
+
+## Build
+
+Build it with colcon build:
+```bash
+cd ~/ros2_ws
+source /opt/ros/humble/setup.bash
+colcon build --packages-up-to ardupilot_cartographer ardupilot_gz_bringup
+
+```
+
+## Usage
+
+### 1. Cartographer running with LiDAR on copter
+
+This simulation has an Iris copter equipped with a 360 degrees 2D LiDAR in a maze world.
+To launch rviz and gazebo, run:
+
+```bash
+cd ~/ros2_ws
+source install/setup.bash
+ros2 launch ardupilot_gz_bringup iris_maze.launch.py
+```
+In another terminal, with the world and copter in place, launch cartographer to generate SLAM:
+
+```bash
+cd ~/ros2_ws
+source install/setup.bash
+ros2 launch ardupilot_cartographer cartographer.launch.py
```

Comment:
Can we avoid directly duplicating the README contents?
Perhaps try to keep the cartographer specific stuff here, and then just refer to the cartographer directory from top level.

Reply:
Got it

Reply:
I kept system requirements and overall build part in top level directory and instead of details on cartographer, I referred to it from that. In cartographer directory, I kept usage part only, referring to setup in parent's readme.